ozaria/site/components/play/PageIntroLevel/index.vue

```diff
@@ -69,163 +61,137 @@
-          this.courseId = this.courseId || utils.getQueryVariable('course')
-          this.campaignId = this.campaignId || utils.getQueryVariable('campaign')
-          try {
-            this.introLevelData = await api.levels.getByIdOrSlug(this.introLevelIdOrSlug)
-            if (me.isSessionless()) { // not saving progress/session for teachers
-              this.language = this.codeLanguage || defaultCodeLanguage
-            } else {
-              this.introLevelSession = await api.levels.upsertSession(this.introLevelIdOrSlug, { courseInstanceId: this.courseInstanceId })
-              this.language = this.introLevelSession.codeLanguage
-            }
-
-            this.introContent = this.introLevelData.introContent
+    methods: {
+      loadIntroLevel: async function () {
+        this.dataLoaded = false
 
-            // fetch campaign data
-            await this.loadCampaignData()
-          } catch (err) {
-            console.error('Error in creating data for intro level', err)
-            // TODO handle_error_ozaria
-            noty({ text: 'Error in creating data for intro level', type: 'error', timeout: 2000 })
-            return
+        // Reading query params because this is rendered via backbone router and cannot be directly passed in as props
+        // They need to be in a specific order in the url to read and send them as props directy from backbone router, hence using query params here.
+        this.courseInstanceId = this.courseInstanceId || utils.getQueryVariable('course-instance')
+        this.codeLanguage = this.codeLanguage || utils.getQueryVariable('code-language')
+        this.courseId = this.courseId || utils.getQueryVariable('course')
+        try {
+          this.introLevelData = await api.levels.getByIdOrSlug(this.introLevelIdOrSlug)
+          if (me.isSessionless()) { // not saving progress/session for teachers
+            this.language = this.codeLanguage || defaultCodeLanguage
+          } else {
+            this.introLevelSession = await api.levels.upsertSession(this.introLevelIdOrSlug, { courseInstanceId: this.courseInstanceId })
+            this.language = this.introLevelSession.codeLanguage
           }
-          // Assign first content in the sequence to this.currentContent
-          this.currentIndex = 0
+
+          this.introContent = this.introLevelData.introContent
+        } catch (err) {
+          console.error('Error in creating data for intro level', err)
+          // TODO handle_error_ozaria
+          noty({ text: 'Error in creating data for intro level', type: 'error', timeout: 2000 })
+          return
+        }
+        // Assign first content in the sequence to this.currentContent
+        this.currentIndex = 0
+        this.currentContent = this.introContent[this.currentIndex]
+        this.setCurrentContentId(this.currentContent)
+        this.dataLoaded = true
+      },
+      onContentCompleted: async function (data) {
+        this.currentContentData = data || {}
```

Comment:
I am a bit confused about this addition, and the cinematic emitting data in the completed event. Would you be able to elaborate what the purpose is?

What is the `this.currentContentData.contentType = this.currentContent.type` line doing?

Reply:
We need the data of the content to know its name for victory modals, and potentially other details in the future.
Cinematics, as well as other systems, are emitting that data with the completed event so that we do not need to call the api again to fetch data based on id/slug.
And then, `this.currentContentData.contentType = this.currentContent.type` adds the type cinematic/interactive/cutscene to that data because its needed for the victory modal.